### PR TITLE
feat: make readme template header validation slightly more lenient

### PR DESCRIPTION
**Pull request by cprice404**
This commit relaxes the template header validation just a bit.
It will still enforce that you have all of the required headers
in the right order, but it will now tolerant cases where you have
additional headers inserted anywhere in the template.  This allows
us to add special sections in individual language repos for cases
where the language requires some additional info that may not be
relevant to the other languages.

We also add a new required `Examples` header to make sure that
we call that out clearly a little earlier in the README.

This is to conform with the latest changes to the .NET README, which
are based on feedback we got during the bug bash.

### CODE REVIEW

File: github-actions/oss-readme-template/__tests__/readme.test.ts [hunk 1]
```diff
@@ -213,10 +223,88 @@ This is an official Momento SDK, but the API is in a beta stage.  For more info,
...
+<img src="https://docs.momentohq.com/img/logo.svg" alt="logo" width="400"/>
+
+[![project status](https://momentohq.github.io/standards-and-practices/badges/project-status-official.svg)](https://github.com/momentohq/standards-and-practices/blob/main/docs/momento-on-github.md)
+[![project stability](https://momentohq.github.io/standards-and-practices/badges/project-stability-stable.svg)](https://github.com/momentohq/standards-and-practices/blob/main/docs/momento-on-github.md) 
+
+# Momento WaterLoop Client Library
+
+
+WaterLoop client SDK for Momento Serverless Cache: a fast, simple, pay-as-you-go caching solution without
+any of the operational overhead required by traditional caching solutions!
```
Comment (malandis) [inline github-actions/oss-readme-template/__tests__/readme.test.ts:260]: Not a blocker for the PR but would help maintainability to externalize expected generated output to separate files

File: github-actions/oss-readme-template/dist/index.js [hunk 2]
```diff
@@ -212,40 +216,41 @@ exports.verifySdkSectionHeaders = {
...
         ];
+        let lastSuccessfulMatch = '';
         for (const expectedHeader of expectedRemainingHeaders) {
-            const nextActualHeader = headers.shift();
+            let nextActualHeader = headers.shift();
             console.info(`Checking for expected header: '${expectedHeader.content}'`);
             console.info(`Next actual header: '${JSON.stringify(nextActualHeader)}'`);
+            while (nextActualHeader !== undefined) {
+                if (expectedHeader.content !== nextActualHeader.content) {
+                    console.info(`Found extra, non-required header: '${nextActualHeader.content}'.  Ignoring.`);
```
Comment (malandis) [inline github-actions/oss-readme-template/dist/index.js:226]: Could also be a duplicate of a required header

**Review by malandis — APPROVED** [review]

File: github-actions/oss-readme-template/dist/index.js (hunk 2 above)
Comment (cprice404) [reply github-actions/oss-readme-template/dist/index.js:226]: That's true.  We can update it if it proves to be confusing.

File: github-actions/oss-readme-template/__tests__/readme.test.ts (hunk 1 above)
Comment (cprice404) [reply github-actions/oss-readme-template/__tests__/readme.test.ts:260]: yeah that's a good point, next time i'm in here I'll improve it.

**Pull request merged**